rust/benchmarks/src/lib.rs

```diff
@@ -54,12 +57,14 @@ impl RunArgs {
     }
 }
 
-pub fn run(args: &RunArgs) {
+pub fn run(args: &mut RunArgs) {
```

Comment:
Can we experiment with this? 
Maybe another grouping removes the mutability of the arguments to scope it only to the context. 
maybe 2 arguments, or instance being self?  wdyt?